### PR TITLE
Syntax editor

### PR DESCRIPTION
**Pull request by makiam**
Common script editors code moved to base class. Old Editor control replaced with new RSyntaxTextArea.
Added new library jar

### CODE REVIEW

File: ArtOfIllusion/src/artofillusion/script/ScriptWindowBase.java [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+import javax.swing.filechooser.FileNameExtensionFilter;
+import org.fife.ui.rsyntaxtextarea.RSyntaxTextArea;
+import org.fife.ui.rsyntaxtextarea.SyntaxConstants;
+import org.fife.ui.rtextarea.RTextScrollPane;
+
+/**
+ *
+ * @author MaksK
+ */
+public class ScriptWindowBase extends BFrame
```
Comment (Sailsman63) [inline ArtOfIllusion/src/artofillusion/script/ScriptWindowBase.java:40]: Not a fan of this approach. As originally implemented, the script editor is simply a widget that provides the text editor functions. It composes well with any window arrangement that you want to embed it in.

As you've written this, one has to extend the `ScriptWindow`. Favor composition over inheritance.

File: ArtOfIllusion/src/artofillusion/script/ScriptedObjectEditorWindow.java [hunk 2]
```diff
@@ -22,7 +22,7 @@ Changes copyright (C) 2020 by Maksim Khramov
 
 /** This class presents a user interface for entering object scripts. */
 
-public class ScriptedObjectEditorWindow extends ScriptWindowBase
+public final class ScriptedObjectEditorWindow extends ScriptWindowBase
```
Comment (Sailsman63) [inline ArtOfIllusion/src/artofillusion/script/ScriptedObjectEditorWindow.java:25]: Why the change to final?

File: ArtOfIllusion/src/artofillusion/script/ExecuteScriptWindow.java [hunk 3]
```diff
@@ -18,7 +18,7 @@ Changes copyright (C) 2020 by Maksim Khramov
 
 /** This class presents a user interface for entering scripts to be executed. */
 
-public class ExecuteScriptWindow extends ScriptWindowBase
+public final class ExecuteScriptWindow extends ScriptWindowBase
```
Comment (Sailsman63) [inline ArtOfIllusion/src/artofillusion/script/ExecuteScriptWindow.java:21]: And here?

File: ArtOfIllusion/src/artofillusion/script/ScriptedObjectEditorWindow.java (hunk 2 above)
Comment (makiam) [reply ArtOfIllusion/src/artofillusion/script/ScriptedObjectEditorWindow.java:25]: This way (Plus ScriptWindowBase is marked as package-private) I prevent unintended extension  this classes.

Comment (Sailsman63) [reply ArtOfIllusion/src/artofillusion/script/ScriptedObjectEditorWindow.java:25]: Again, why?

If this is in response to the concern about the structure that I expressed the first time through, you seem to have misunderstood me...

Comment (makiam) [reply ArtOfIllusion/src/artofillusion/script/ScriptedObjectEditorWindow.java:25]: Just not found good solution with composition style against inheritance style

Comment (peastman) [reply ArtOfIllusion/src/artofillusion/script/ScriptedObjectEditorWindow.java:25]: Why do you want to prevent people from extending it?

The point was that if the editor is a widget, anyone can embed an editor inside any window.  But if the editor is a window, it can only be used in that window and no other.

Comment (makiam) [reply ArtOfIllusion/src/artofillusion/script/ScriptedObjectEditorWindow.java:25]: As @Sailsman63 wrote

> As you've written this, one has to extend the ScriptWindow

I close this views (Script editor and ScriptObject editor) from extending nd reduce visibility of their base class. 
... Maybe I do not understand the idea in general

Comment (Sailsman63) [reply ArtOfIllusion/src/artofillusion/script/ScriptedObjectEditorWindow.java:25]: I can run a PR against your branch to show you what I mean. It may be a few days.
